internal/blocktx/processor_helpers.go

```diff
@@ -0,0 +1,110 @@
+package blocktx
+
+import (
+	"encoding/binary"
+	"math/big"
+
+	"github.com/bitcoin-sv/arc/internal/blocktx/blocktx_api"
+	sdkTx "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/libsv/go-p2p"
+)
+
+// exported for testing purposes
+func extractHeightFromCoinbaseTx(tx *sdkTx.Transaction) uint64 {
+	// Coinbase tx has a special format, the height is encoded in the first 4 bytes of the scriptSig
+	// https://en.bitcoin.it/wiki/Protocol_documentation#tx
+	// Get the length
+	script := *(tx.Inputs[0].UnlockingScript)
+	length := int(script[0])
+
+	if len(script) < length+1 {
+		return 0
+	}
+
+	b := make([]byte, 8)
+
+	for i := 0; i < length; i++ {
+		b[i] = script[i+1]
+	}
+
+	return binary.LittleEndian.Uint64(b)
+}
+
+func createBlock(msg *p2p.BlockMessage, prevBlock *blocktx_api.Block, longestTipExists bool) *blocktx_api.Block {
+	hash := msg.Header.BlockHash()
+	prevHash := msg.Header.PrevBlock
+	merkleRoot := msg.Header.MerkleRoot
+	chainwork := calculateChainwork(msg.Header.Bits)
+
+	var status blocktx_api.Status
+	if prevBlock == nil {
+		if longestTipExists {
+			status = blocktx_api.Status_ORPHANED
```

Comment:
That is the extreme edge case that the first block already has a competing block, right?
Wouldn't it be `Status_STALE` then? 🤔

Reply:
No no, this is for the edge case that there is no block in the database, it's empty.

Then, the first block that comes will be marked as `ORPHANED` because it doesn't have any previous blocks and then every next block will also be marked as `ORPHANED` because it's extending the orphaned chain.

To avoid that situation, I'm checking if there is any block in the db (by using the getTip method) and if there isn't, then the incoming block is `LONGEST`. If there is, then the incoming block (without the prevBlock in our db) will be marked as `ORPHANED`, correctly.

Please let me know if I explained it well

Reply:
Also, I left the further handling of ORPHANED blocks out of scope of this PR. I should probably create a ticket for that.

Reply:
By "further handling" I mean:
1. Removing the `orhpanedyn` field, and using the newly created status.
2. Putting the blocks out of `ORPHANED` state into either `LONGEST` or `STALE` in case the previous block comes to our database